etl/steps/data/garden/terrorism/2023-07-20/global_terrorism_database.py

```diff
@@ -0,0 +1,253 @@
+        - number of people killed
+        - number of people wounded
+        - number of terrorism attacks by target
+        - number of terrorism attacks by weapon type
+        - number of terrorism attacks by attack type
+
+    It then calculates per capita metrics, adds decadal averages, and saves the results in a new garden
+    dataset.
+
+    Parameters:
+        dest_dir (str): The destination directory where the new garden dataset will be saved.
+
+    Returns:
+        None
+    """
+
+    # Load inputs from the meadow dataset.
+    ds_meadow_terrorism = cast(Dataset, paths.load_dependency("global_terrorism_database"))
+    tb_terrorism = ds_meadow_terrorism["global_terrorism_database"]
+    tb_terrorism.reset_index(inplace=True)
+
+    # Process data to calculate statistics related to terrorism incidents.
+    tb: Table = geo.harmonize_countries(df=tb_terrorism, countries_file=paths.country_mapping_path)
+    total_df = pd.DataFrame()
+    total_df["total_killed"] = tb.groupby(["country", "year"])["nkill"].sum()
+    total_df["total_wounded"] = tb.groupby(["country", "year"])["nwound"].sum()
+    total_df["total_incident_counts"] = tb.groupby(["country", "year"]).size()
+    total_df["total_casualties"] = total_df["total_wounded"] + total_df["total_incident_counts"]
+
+    # Process terrorism targets, attacks, and weapons data.
+    pivot_weapon_type, pivot_target_type, pivot_df_attack_type = terrorism_targets_attacks_weapons(tb)
+    merge_weapon_target = pd.merge(pivot_target_type, pivot_weapon_type, on=["country", "year"])
+    merge_attack = pd.merge(pivot_df_attack_type, merge_weapon_target, on=["country", "year"])
+    merge_all = pd.merge(total_df, merge_attack, on=["country", "year"])
+
+    # Add deaths and population data, and region aggregates.
+    df_pop_deaths = add_deaths_and_population(merge_all)
+    ds_regions: Dataset = paths.load_dependency("regions")
+    df_pop_deaths = add_data_for_regions(tb=df_pop_deaths, regions=REGIONS, ds_regions=ds_regions)
+
+    # Calculate statistics per capita
+    df_pop_deaths["terrorism_wounded_per_capita"] = df_pop_deaths["total_wounded"] / df_pop_deaths["population"]
+    df_pop_deaths["terrorism_deaths_per_capita"] = df_pop_deaths["total_killed"] / df_pop_deaths["population"]
+    df_pop_deaths["terrorism_casualties_per_capita"] = df_pop_deaths["total_casualties"] / df_pop_deaths["population"]
+
+    # Perform decadal averaging for selected columns
+    cols_for_decadal_av = [
+        "total_killed",
+        "total_wounded",
+        "total_incident_counts",
+        "total_casualties",
+        "terrorism_wounded_per_capita",
+        "terrorism_deaths_per_capita",
+        "terrorism_casualties_per_capita",
+    ]
+
+    df_pop_deaths = perform_decadal_averaging(df_pop_deaths, cols_for_decadal_av=cols_for_decadal_av)
+
+    # Convert relevant columns to float64 data type (to avoid errors related to this issue -  https://github.com/owid/etl/issues/1334)
```

Comment:
Oh man, working with both `float64` and `Float64` is a nightmare! What we often do is that we convert everything to `float` right after loading to avoid these problems. You could do

```
tb_terrorism = tb_terrorism.astype(
        {
            "nkill": float,
            "nwound": float,
        }
    )
```
it doesn't really matter that these are really not floats, but ints with NaN values. When saving the dataset we optimize types anyway.